contrib/inventory/ec2.py

```diff
@@ -881,6 +882,16 @@ def add_instance(self, instance, region):
             if self.nested_groups:
                 self.push_group(self.inventory, 'instance_states', state_name)
 
+        # Inventory: Group by platform
+        if self.group_by_platform:
+            if instance.platform:
+                platform = self.to_safe('platform_' + instance.platform)
+            else:
+                platform = self.to_safe('platform_undefined')
+            self.push(self.inventory, platform, dest)
```

Comment:
I think this should be hostname, since hostname defaults to dest ([line 825](https://github.com/ansible/ansible/pull/14246/files#diff-c2e371537339bbab5d9d0cc5d6cb7adcL825)). That way this maintains consistency with the surrounding code. Should hostname be defined, I think it might be confusing. That was my only feedback though, this looks good.